presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java

```diff
@@ -28,14 +30,10 @@
 @ThreadSafe
 public class NodeSet
 {
-    private static NodeSet nodeSetInstance = new NodeSet();
-
-    public static Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+    private Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
 
-    public static NodeSet getInstance()
-    {
-        return nodeSetInstance;
-    }
+    @Inject
```

Comment:
Oops.
Then the annotation has no effect at all here. I'd rather leave it to minimize conflicts with the `discovery` branch, where something _does_ get injected.